terraform_validate.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Function to run terraform init and validate in a directory
+run_terraform() {
+    local dir="$1"
+    echo "Running terraform init and validate in $dir"
+    cd "$dir" || exit
+    terraform init -upgrade && terraform validate
+    cd - || exit
```

Comment:
We're already using `set -e`, no need for `&&` or `||`.
```suggestion
    pushd "$dir"
    terraform init -upgrade
    terraform validate
    popd
```